cloud/amazon/cloudfront.py

```diff
@@ -0,0 +1,557 @@
+
+    invalidations = client.list_invalidations(DistributionId=distribution_id)
+
+    invalidation_configs = []
+    if 'Items' in invalidations['InvalidationList']:
+        for invalidation in invalidations['InvalidationList']['Items']:
+            temp_object = dict()
+            config = client.get_invalidation(
+                DistributionId=distribution_id, Id=invalidation['Id'])
+            temp_object['Config'] = config['Invalidation']['InvalidationBatch']
+            temp_object['ETag'] = ""
+            temp_object['Id'] = invalidation['Id']
+            invalidation_configs.append(temp_object)
+    return invalidation_configs
+
+
+def get_existing_streaming_distributions(client, module):
+    distributions = client.list_streaming_distributions()
+    dist_configs = []
+    for dist in distributions['StreamingDistributionList']['Items']:
+        temp_object = dict()
+        config = client.get_streaming_distribution_config(
+            Id=dist['Id'])
+        temp_object['Config'] = config['StreamingDistributionConfig']
+        temp_object['ETag'] = config['ETag']
+        temp_object['Id'] = dist['Id']
+        temp_object['DomainName'] = dist['DomainName']
+        dist_configs.append(temp_object)
+    return dist_configs
+
+
+def creation_setup(client, module):
+    policy = None
+    changed = False
+
+    if module.params.get('policy'):
+        try:
+            with open(module.params.get('policy')) as json_data:
+                try:
+                    policy = json.load(json_data)
+                    json_data.close()
+                except ValueError as e:
+                    module.fail_json(msg=str(e))
+        except (OSError, IOError) as e:
+            try:
+                policy = json.loads(module.params.get('policy'))
+            except ValueError as e:
+                module.fail_json(msg=str(e))
+        except Exception as e:
+            module.fail_json(msg=str(e))
+
+    invocations = {
+        'distribution': {
+            'get_config_method': get_existing_distributions,
+        },
+        'origin_access_id': {
+            'get_config_method': get_existing_origin_access_id,
+        },
+        'invalidation': {
+            'get_config_method': get_existing_invalidations,
+        },
+        'streaming': {
+            'get_config_method': get_existing_streaming_distributions
+        }
+    }
+
+    if 'CallerReference' not in policy:
+        module.fail_json(msg='CallerReference is required in your policy')
+    else:
+        caller_reference = policy['CallerReference']
+
+    resource_type = invocations[module.params.get('type')]
+    existing = resource_type['get_config_method'](client, module)
+    caller_reference_exists = False
+    for item in existing:
+        if caller_reference == item['Config']['CallerReference']:
+            caller_reference_exists = True
+            if cmp(policy, item['Config']) == 0:
```

Comment:
Why wouldn't you just do `policy == item['Config']` ? It's the same, `cmp` just returns 0 if `a == b`